src/ln/msgs.rs

```diff
@@ -585,6 +586,9 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn handle_channel_update(&self, msg: &ChannelUpdate) -> Result<bool, HandleError>;
 	/// Handle some updates to the route graph that we learned due to an outbound failed payment.
 	fn handle_htlc_fail_channel_update(&self, update: &HTLCFailChannelUpdate);
+	///Gets a subset of the messages required to dump our routing table to a remote node, starting at the short_channel_id indicated by starting_point and including batch_amount entries
+	/// This function will start iterating at 0 if the starting_point is < 0.
+	fn get_next_announcements(&self, starting_point: u64, batch_amount: u8)->(Vec<(ChannelAnnouncement, NodeAnnouncement, NodeAnnouncement,ChannelUpdate,ChannelUpdate)>, i64);
```

Comment:
Thinking about this a little more, while its technically ok, it really sucks to send so many duplicate NodeAnnouncements here. Maybe split it into two steps - the first does ChannelAnnouncement/ChannelUpdate, then do a second pass that just does NodeAnnouncements. This would mean swapping last_synced_channel for an enum { A(PublicKey), B(u64), C}, but that'd be cleaner anyway, even if a bit more memory.

Reply:
split them up, I dont think it will really take up more memory thou..

Reply:
Not about memory, its about double-sending. Because a node can have many channels associated with it sending a node_announcement once for each channel kinda sucks.